src/App.tsx

```diff
@@ -1,223 +1,170 @@
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/Error/ErrorNotification';
+import { TodoList } from './components/TodoList/TodoList';
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoCount } from './components/TodoCount/TodoCount';
+import { TodoFilter } from './components/Filter/TodoFilter';
+import { FilterBy } from './types/FilterType';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortBy, setSortBy] = useState(FilterBy.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [isDeleting, setIsDeleting] = useState(false);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    newTodoField.current?.focus();
+  }, [todos]);
+
+  useEffect(() => {
+    getTodos(user?.id || 0)
+      .then(todosFromServer => setTodos(todosFromServer))
+      .catch(() => setErrorMessage('Unable to load data'));
+  }, [user]);
+
+  const visibleTodo = todos.filter((todo) => {
+    switch (sortBy) {
+      case FilterBy.All:
+        return todo;
+
+      case FilterBy.Active:
+        return !todo.completed;
+
+      case FilterBy.Completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
+  const activeTodo = todos.filter((todo) => {
+    return !todo.completed;
+  });
+
+  const completedTodo = todos.filter((todo) => {
+    return todo.completed;
+  });
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setErrorMessage('');
+
+    setNewTodoTitle(value);
+  };
+
+  const handleAddNewTodo = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!newTodoTitle.trim()) {
+      setErrorMessage('Title can\'t be empty');
+
+      return;
+    }
+
+    setIsAdding(true);
+    try {
+      const newTodo = await addTodo(newTodoTitle, user?.id || 0);
+
+      setTodos(prevTodos => [...prevTodos, newTodo]);
+      setNewTodoTitle('');
+    } catch {
+      setErrorMessage('Unable to add a todo');
+    }
+
+    setIsAdding(false);
+  };
+
+  const handleDeleteTodo = async (todoId: number) => {
+    setIsDeleting(true);
+    try {
+      await deleteTodo(todoId);
+
+      setTodos([...visibleTodo].filter((todo) => todo.id !== todoId));
+    } catch {
+      setErrorMessage('Unable to delete a todo');
+    }
+
+    setIsDeleting(false);
```

Comment:
you are set your isDeleting as true, start async action and set it fals like in 5ms after first setIsDeleting